Lib/test/test_mmap.py

```diff
@@ -1058,6 +1060,59 @@ def __exit__(self, exc_type, exc_value, traceback):
                 with self.assertRaisesRegex(ValueError, "mmap closed or invalid"):
                     m.write_byte(X())
 
+    @unittest.skipUnless(os.name == 'nt', 'requires Windows')
+    @unittest.skipUnless(hasattr(mmap.mmap, '_protect'), 'test needs debug build')
+    def test_access_violations(self):
+    
+        code = textwrap.dedent("""
+            from test.support.os_helper import TESTFN
+            import mmap
+            import os
+            from contextlib import suppress
+
+            if os.path.exists(TESTFN):
+                os.unlink(TESTFN)
+
+            PAGESIZE = mmap.PAGESIZE
+            PAGE_NOACCESS = 0x01
+
+            with open(TESTFN, 'bw+') as f:
+                f.write(b'A'* PAGESIZE)
+                f.flush()
+
+                m = mmap.mmap(f.fileno(), PAGESIZE)
+                m._protect(PAGE_NOACCESS, 0, PAGESIZE)
+                with suppress(OSError):
+                    m.read(PAGESIZE)
+                with suppress(OSError):
+                    m.read_byte()
+                with suppress(OSError):
+                    m.readline()
+                with suppress(OSError):
+                    m.write(b'A'* PAGESIZE)
+                with suppress(OSError):
+                    m.write_byte(0)
+                with suppress(OSError):
+                    m[0]  # test mmap_subscript
+                with suppress(OSError):
+                    m[0:10]  # test mmap_subscript
+                with suppress(OSError):
+                    m[0:10:2]  # test mmap_subscript
+                with suppress(OSError):
+                    m[0] = 1
+                with suppress(OSError):
+                    m[0:10] = b'A'* 10
+                with suppress(OSError):
+                    m[0:10:2] = b'A'* 5
+                with suppress(OSError):
+                    m.move(0, 10, 1)
+                with suppress(OSError):
+                    list(m)  # test mmap_item
+        """)
+        # stderr includes "Windows fatal exception: access violation" logs,
+        # even when the access violations are correctly transformed into exceptions
```

Comment:
Call `faulthandler.disable()` at the start of the executed code. The issue is that mmap uses a stack handler, and vectored handlers, such as the one set by faulthandler, are called first.